src/modm/board/devebox_stm32h750vb/board.hpp

```diff
@@ -0,0 +1,172 @@
+	static constexpr uint32_t Ahb4 = Ahb;
+	// Max 100MHz or 120MHz
+	static constexpr uint32_t Apb1 = Ahb / 2; // D2PPRE1
+	static constexpr uint32_t Apb2 = Ahb / 2; // D2PPRE2
+	static constexpr uint32_t Apb3 = Ahb / 2; // D1PPRE
+	static constexpr uint32_t Apb4 = Ahb / 2; // D3PPRE
+
+	static constexpr uint32_t Adc1 = Ahb1;
+	static constexpr uint32_t Adc2 = Ahb1;
+	static constexpr uint32_t Adc3 = Ahb4;
+
+	static constexpr uint32_t Dac1 = Apb1;
+
+	static constexpr uint32_t Spi1 = Apb2;
+	static constexpr uint32_t Spi2 = Apb1;
+	static constexpr uint32_t Spi3 = Apb1;
+	static constexpr uint32_t Spi4 = Apb2;
+	static constexpr uint32_t Spi5 = Apb2;
+	static constexpr uint32_t Spi6 = Apb4;
+
+	static constexpr uint32_t Usart1 = Apb2;
+	static constexpr uint32_t Usart2 = Apb1;
+	static constexpr uint32_t Usart3 = Apb1;
+	static constexpr uint32_t Uart4  = Apb1;
+	static constexpr uint32_t Uart5  = Apb1;
+	static constexpr uint32_t Usart6 = Apb2;
+	static constexpr uint32_t Uart7  = Apb1;
+	static constexpr uint32_t Uart8  = Apb1;
+
+	static constexpr uint32_t LpUart1 = Apb4;
+
+	static constexpr uint32_t Can1 = Apb1;
+	static constexpr uint32_t Can2 = Apb1;
+
+	static constexpr uint32_t I2c1 = Apb1;
+	static constexpr uint32_t I2c2 = Apb1;
+	static constexpr uint32_t I2c3 = Apb1;
+	static constexpr uint32_t I2c4 = Apb4;
+
+	static constexpr uint32_t Apb1Timer = Apb1 * 2;
+	static constexpr uint32_t Apb2Timer = Apb2 * 2;
+	static constexpr uint32_t Timer1  = Apb2Timer;
+	static constexpr uint32_t Timer2  = Apb1Timer;
+	static constexpr uint32_t Timer3  = Apb1Timer;
+	static constexpr uint32_t Timer4  = Apb1Timer;
+	static constexpr uint32_t Timer5  = Apb1Timer;
+	static constexpr uint32_t Timer6  = Apb1Timer;
+	static constexpr uint32_t Timer7  = Apb1Timer;
+	static constexpr uint32_t Timer8  = Apb2Timer;
+	static constexpr uint32_t Timer12 = Apb1Timer;
+	static constexpr uint32_t Timer13 = Apb1Timer;
+	static constexpr uint32_t Timer14 = Apb1Timer;
+	static constexpr uint32_t Timer15 = Apb2Timer;
+	static constexpr uint32_t Timer16 = Apb2Timer;
+	static constexpr uint32_t Timer17 = Apb2Timer;
+
+	static constexpr uint32_t Usb = 50_MHz; // From PLL3Q
+
+	static bool inline
+	enable()
+	{
+		Rcc::enableExternalCrystal(); // 25MHz
+		Rcc::setVoltageScaling(Rcc::VoltageScaling::Scale0); // required for 400MHz/480MHz
+		const Rcc::PllFactors pllFactors1{
+			.range = Rcc::PllInputRange::MHz8_16,
+			.pllM = 2,		//   25MHz / M= 12.5MHz
+			.pllN = 64,		// 12.5MHz * N=  800MHz
+			.pllP = 2,		//  800MHz / P=  400MHz = F_cpu
+			.pllQ = 20,		//  800MHz / Q=   40MHz
+			.pllR = 20,		//  800MHz / R=   40MHz
+		};
+		Rcc::enablePll1(Rcc::PllSource::ExternalClock, pllFactors1);
+		// Use PLL3 for USB 50MHz
+		const Rcc::PllFactors pllFactors3{
+			.range = Rcc::PllInputRange::MHz8_16,
+			.pllM = 2,		//   25MHz / M=  12.5MHz
+			.pllN = 40,		// 12.5MHz * N=   500MHz
+			.pllP = 2,		//  500MHz / P=   250MHz
+			.pllQ = 10,		//  500MHz / Q=    50MHz = F_usb
```

Comment:
FYI: This should be 48MHz, but it's kinda hard to get that from 25MHz… The PLL has a fractional mode, however it's not implemented yet. Also I couldn't get TinyUSB to work on my H743, so there's probably more broken than just the clock.

Reply:
It's possible to get it by dividing by 5 twice and multiplying by 48 (M=5, N=48, Q=5), CubeMX says it should work.

Reply:
@chris-durand of cause haha. I somehow missed that